File: Lynx-IOS/Lynx-IOS/Features/Chat/ChatInputBar.swift
import SwiftUI

/// Mobile version of the "terminal chrome" input from v0-ai-chat.tsx:
/// amber corner marks, auto-growing text field, amber send button.
struct ChatInputBar: View {
    @Binding var text: String
    var placeholder: String = "Describe what you want to build…"
    var onSend: () -> Void

    @FocusState private var isFocused: Bool

    private var canSend: Bool { !text.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty }

    var body: some View {
        VStack(spacing: 0) {
            TextField("", text: $text, prompt: Text(placeholder).foregroundStyle(LynxColor.stone600), axis: .vertical)
                .font(LynxFont.mono(14))
                .foregroundStyle(LynxColor.stone200)
                .tint(LynxColor.amberBright)
                .lineLimit(1...6)
                .focused($isFocused)
                .padding(.horizontal, 14)
                .padding(.vertical, 14)
                .onSubmit(send)

            HStack {
                Button {} label: {
                    Image(systemName: "paperclip")
                        .foregroundStyle(LynxColor.stone600)
                }
                Spacer()
                Button(action: send) {
                    Image(systemName: "arrow.up")
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundStyle(canSend ? .black : LynxColor.stone600)
                        .frame(width: 28, height: 28)
                        .background(canSend ? LynxColor.amberBright : .clear)
                        .overlay(
                            RoundedRectangle(cornerRadius: LynxMetrics.cornerRadius)
                                .strokeBorder(canSend ? .clear : LynxColor.stone700, lineWidth: 1)
                        )
                        .clipShape(RoundedRectangle(cornerRadius: LynxMetrics.cornerRadius))
                }
                .disabled(!canSend)
            }
            .padding(.horizontal, 12)
            .padding(.bottom, 10)
            .padding(.top, 4)
            .overlay(alignment: .top) {
                Rectangle().fill(LynxColor.stone800.opacity(0.8)).frame(height: 1)
            }
        }
        .background(
            RoundedRectangle(cornerRadius: LynxMetrics.cornerRadius)
                .fill(LynxColor.panelBg)
                .overlay(
                    RoundedRectangle(cornerRadius: LynxMetrics.cornerRadius)
                        .strokeBorder(LynxColor.stone800, lineWidth: 1)
                )
        )
        // .overlay proposes the base view's own (now content-sized) frame to
        // the GeometryReader below, instead of a ZStack sibling's unbounded one.
        .overlay(cornerMarks)
        .clipShape(RoundedRectangle(cornerRadius: LynxMetrics.cornerRadius))
    }

    private func send() {
        guard canSend else { return }
        onSend()
    }

    private var cornerMarks: some View {
        GeometryReader { geo in
            let mark: CGFloat = 10
            ForEach(0..<4, id: \.self) { corner in
                Path { path in
                    let isTop = corner < 2
                    let isLeft = corner % 2 == 0
                    let x: CGFloat = isLeft ? 0 : geo.size.width
                    let y: CGFloat = isTop ? 0 : geo.size.height
                    let dx: CGFloat = isLeft ? mark : -mark
                    let dy: CGFloat = isTop ? mark : -mark
                    path.move(to: CGPoint(x: x, y: y + dy))
                    path.addLine(to: CGPoint(x: x, y: y))
                    path.addLine(to: CGPoint(x: x + dx, y: y))
                }
                .stroke(LynxColor.amber.opacity(0.6), lineWidth: 1)
            }
        }
        .allowsHitTesting(false)
    }
}

File: Lynx-Andrioid/app/src/main/java/com/anantha/lynx/ui/chat/ChatInputBar.kt
package com.anantha.lynx.ui.chat

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Send
import androidx.compose.material.icons.filled.AttachFile
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.anantha.lynx.ui.theme.LynxColor
import com.anantha.lynx.ui.theme.LynxFont
import com.anantha.lynx.ui.theme.LynxMetrics

/**
 * Ported from Lynx-IOS/Lynx-IOS/Features/Chat/ChatInputBar.swift — auto-
 * growing text field with amber corner-bracket "terminal chrome" decoration.
 */
@Composable
fun ChatInputBar(
    text: String,
    onTextChange: (String) -> Unit,
    onSend: () -> Unit,
    modifier: Modifier = Modifier,
    placeholder: String = "Describe what you want to build…",
) {
    val canSend = text.isNotBlank()

    Box(
        modifier = modifier
            .background(LynxColor.panelBg, RoundedCornerShape(LynxMetrics.cornerRadius))
            .border(1.dp, LynxColor.stone800, RoundedCornerShape(LynxMetrics.cornerRadius)),
    ) {
        Column {
            Box(modifier = Modifier.padding(horizontal = 14.dp, vertical = 14.dp)) {
                BasicTextField(
                    value = text,
                    onValueChange = onTextChange,
                    textStyle = LynxFont.mono(14.sp).copy(color = LynxColor.stone200),
                    cursorBrush = androidx.compose.ui.graphics.SolidColor(LynxColor.amberBright),
                    modifier = Modifier.fillMaxWidth().heightIn(min = 20.dp, max = 120.dp),
                    decorationBox = { innerTextField ->
                        if (text.isEmpty()) {
                            Text(placeholder, style = LynxFont.mono(14.sp), color = LynxColor.stone600)
                        }
                        innerTextField()
                    },
                )
            }

            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 12.dp)
                    .padding(top = 4.dp, bottom = 10.dp),
            ) {
                IconButton(onClick = { /* no-op, mirrors iOS placeholder button */ }) {
                    Icon(Icons.Filled.AttachFile, contentDescription = "Attach", tint = LynxColor.stone600)
                }
                Spacer(modifier = Modifier.weight(1f))
                Box(
                    modifier = Modifier
                        .size(28.dp)
                        .background(
                            if (canSend) LynxColor.amberBright else Color.Transparent,
                            RoundedCornerShape(LynxMetrics.cornerRadius),
                        )
                        .border(
                            width = if (canSend) 0.dp else 1.dp,
                            color = LynxColor.stone700,
                            shape = RoundedCornerShape(LynxMetrics.cornerRadius),
                        ),
                    contentAlignment = Alignment.Center,
                ) {
                    IconButton(onClick = { if (canSend) onSend() }, enabled = canSend) {
                        Icon(
                            Icons.AutoMirrored.Filled.Send,
                            contentDescription = "Send",
                            tint = if (canSend) Color.Black else LynxColor.stone600,
                            modifier = Modifier.size(14.dp),
                        )
                    }
                }
            }
        }

        CornerMarks(modifier = Modifier.matchParentSize())
    }
}

@Composable
private fun CornerMarks(modifier: Modifier = Modifier) {
    Canvas(modifier = modifier) {
        val mark = 10.dp.toPx()
        val stroke = Stroke(width = 1.dp.toPx())
        val color = LynxColor.amber.copy(alpha = 0.6f)
        for (corner in 0 until 4) {
            val isTop = corner < 2
            val isLeft = corner % 2 == 0
            val x = if (isLeft) 0f else size.width
            val y = if (isTop) 0f else size.height
            val dx = if (isLeft) mark else -mark
            val dy = if (isTop) mark else -mark
            val path = Path().apply {
                moveTo(x, y + dy)
                lineTo(x, y)
                lineTo(x + dx, y)
            }
            drawPath(path, color = color, style = stroke)
        }
    }
}
